packages/@n8n/api-types/src/dto/credentials/credentials-get-many-request.dto.ts

```diff
@@ -0,0 +1,22 @@
+import { Z } from 'zod-class';
+
+import { booleanLiteral } from 'dto/common';
+
+export class CredentialsGetManyRequestQuery extends Z.class({
```

Comment:
quick test suite for this:

```typescript
import { CredentialsGetManyRequestQuery } from '../credentials-get-many-request.dto';

describe('CredentialsGetManyRequestQuery', () => {
	describe('should pass validation', () => {
		it('with empty object', () => {
			const data = {};

			const result = CredentialsGetManyRequestQuery.safeParse(data);

			expect(result.success).toBe(true);
		});

		test.each([
			{ field: 'includeScopes', value: 'true' },
			{ field: 'includeScopes', value: 'false' },
			{ field: 'includeData', value: 'true' },
			{ field: 'includeData', value: 'false' },
		])('with $field set to $value', ({ field, value }) => {
			const data = { [field]: value };

			const result = CredentialsGetManyRequestQuery.safeParse(data);

			expect(result.success).toBe(true);
		});

		it('with both parameters set', () => {
			const data = {
				includeScopes: 'true',
				includeData: 'true',
			};

			const result = CredentialsGetManyRequestQuery.safeParse(data);

			expect(result.success).toBe(true);
		});
	});

	describe('should fail validation', () => {
		test.each([
			{ field: 'includeScopes', value: true },
			{ field: 'includeScopes', value: false },
			{ field: 'includeScopes', value: 'invalid' },
			{ field: 'includeData', value: true },
			{ field: 'includeData', value: false },
			{ field: 'includeData', value: 'invalid' },
		])('with invalid value $value for $field', ({ field, value }) => {
			const data = { [field]: value };

			const result = CredentialsGetManyRequestQuery.safeParse(data);

			expect(result.success).toBe(false);
			expect(result.error?.issues[0].path[0]).toBe(field);
		});
	})
});

```

Reply:
Interesting, it seems `safeParse` can throw when using `zod-class`:

![image](https://github.com/user-attachments/assets/08d05376-08af-4d98-af1c-24a737fb2a1f)

Reply:
Yeah, using `z.object` works as expected. :/

Reply:
Yeah, transforms don't work when using `zod-class`. 

TS complains if you do this:


```ts
			const instance = new CredentialsGetManyRequestQuery({ includeScopes: 'true' });
```

And you get a runtime error if you do this:

```ts
			const instance = new CredentialsGetManyRequestQuery({ includeScopes: true });
```

@netroy how about we drop `zod-class` and just exports the types manually using `z.output<>`?

Reply:
I'd be fine dropping `zod-class`. The only issue is that decorator metadata only works with concrete runtime types, and not with compile-time types. So, we'd either need another way to wrap these schemas into classes, or find an alternative to decorator metadata for the auto-injection mechanism to work.

Reply:
maybe an intermediate solution would be to fix [this](https://github.com/sam-goodwin/zod-class/issues/31).

Reply:
Like this:

```ts
	async getMany(
		req: CredentialRequest.GetMany,
		_res: unknown,
		@Query(CredentialsGetManyRequestQuery) query: z.output<typeof CredentialsGetManyRequestQuery>,
	) {
```

Maybe there is also a way to make it infer the type somehow.

Reply:
Or we leave this as an exercise for another day.

Reply:
Just saw your comments, they didn't show up before. Probably had to refresh. Let me check the issue.